ntoskrnl/se/debug.c

```diff
@@ -0,0 +1,479 @@
+    }
+
+    if (SdControl & SE_GROUP_DEFAULTED)
+    {
+        DPRINT1("SE_GROUP_DEFAULTED\n");
+    }
+
+    if (SdControl & SE_DACL_PRESENT)
+    {
+        DPRINT1("SE_DACL_PRESENT\n");
+    }
+
+    if (SdControl & SE_DACL_DEFAULTED)
+    {
+        DPRINT1("SE_DACL_DEFAULTED\n");
+    }
+
+    if (SdControl & SE_SACL_PRESENT)
+    {
+        DPRINT1("SE_SACL_PRESENT\n");
+    }
+
+    if (SdControl & SE_SACL_DEFAULTED)
+    {
+        DPRINT1("SE_SACL_DEFAULTED\n");
+    }
+
+    if (SdControl & SE_DACL_UNTRUSTED)
+    {
+        DPRINT1("SE_DACL_UNTRUSTED\n");
+    }
+
+    if (SdControl & SE_SERVER_SECURITY)
+    {
+        DPRINT1("SE_SERVER_SECURITY\n");
+    }
+
+    if (SdControl & SE_DACL_AUTO_INHERIT_REQ)
+    {
+        DPRINT1("SE_DACL_AUTO_INHERIT_REQ\n");
+    }
+
+    if (SdControl & SE_SACL_AUTO_INHERIT_REQ)
+    {
+        DPRINT1("SE_SACL_AUTO_INHERIT_REQ\n");
+    }
+
+    if (SdControl & SE_DACL_AUTO_INHERITED)
+    {
+        DPRINT1("SE_DACL_AUTO_INHERITED\n");
+    }
+
+    if (SdControl & SE_SACL_AUTO_INHERITED)
+    {
+        DPRINT1("SE_SACL_AUTO_INHERITED\n");
+    }
+
+    if (SdControl & SE_DACL_PROTECTED)
+    {
+        DPRINT1("SE_DACL_PROTECTED\n");
+    }
+
+    if (SdControl & SE_SACL_PROTECTED)
+    {
+        DPRINT1("SE_SACL_PROTECTED\n");
+    }
+
+    if (SdControl & SE_RM_CONTROL_VALID)
+    {
+        DPRINT1("SE_RM_CONTROL_VALID\n");
+    }
+
+    if (SdControl & SE_SELF_RELATIVE)
+    {
+        DPRINT1("SE_SELF_RELATIVE\n");
+    }
+}
+
+/**
+ * @brief
+ * Dumps each security identifier (SID)
+ * of an access token to debugger.
+ *
+ * @param[in] Sids
+ * A pointer to an array of SIDs with
+ * associated attributes. The SID is
+ * gathered from such array.
+ *
+ * @param[in] SidCount
+ * The count of SIDs that an access
+ * token has.
+ *
+ * @return
+ * Nothing.
+ */
+static
+VOID
+SepDumpSidsOfToken(
+    _In_ PSID_AND_ATTRIBUTES Sids,
+    _In_ ULONG SidCount)
+{
+    ULONG SidIndex;
+    ULONG SidNumber;
+    UNICODE_STRING SidString;
+
+    /* Loop all SIDs and dump them */
+    SidNumber = 1;
+    for (SidIndex = 0; SidIndex < SidCount; SidIndex++)
+    {
+        RtlConvertSidToUnicodeString(&SidString, Sids[SidIndex].Sid, TRUE);
+        DPRINT1("%lu# %wZ\n", SidNumber, &SidString);
+        RtlFreeUnicodeString(&SidString);
+
+        /* Increment the SID number as we go further */
+        SidNumber++;
+    }
+}
+
+/* PUBLIC FUNCTIONS ***********************************************************/
+
+/**
+ * @brief
+ * Dumps debug information of a security
+ * descriptor to the debugger.
+ *
+ * @param[in] SecurityDescriptor
+ * Pointer to a security descriptor of
+ * which debug data is to be dumped form.
+ *
+ * @return
+ * Nothing.
+ */
+VOID
+SepDumpSdDebugInfo(
+    _In_ PISECURITY_DESCRIPTOR SecurityDescriptor)
+{
+    UNICODE_STRING SidString;
+    PSID OwnerSid, GroupSid;
+    PACL Dacl, Sacl;
+
+    /* Don't dump anything if no SD was provided */
+    if (!SecurityDescriptor)
```

Comment:
Not sure this is useful tbh since your parameter annotation told that the SecurityDescriptor was *not* an optional parameter.